cg/meta/workflow/taxprofiler.py

```diff
@@ -101,24 +112,46 @@ def write_sample_sheet(
                 sample_sheet_content.setdefault(headers, []).extend(contents)
 
             LOG.info(sample_sheet_content)
+            if dry_run:
+                continue
             self.write_sample_sheet_csv(
                 samplesheet_content=sample_sheet_content,
                 headers=TAXPROFILER_SAMPLE_SHEET_HEADERS,
                 config_path=self.get_case_config_path(case_id=case_id),
             )
 
+    def get_pipeline_parameters(self, case_id: str) -> PipelineParameters:
+        """Get taxprofiler parameters."""
+        return TaxprofilerParameters(
+            clusterOptions=f"--qos={self.get_slurm_qos_for_case(case_id=case_id)}",
+            input=self.get_case_config_path(case_id=case_id),
+            outdir=self.get_case_path(case_id=case_id),
+            databases=self.databases,
+            hostremoval_reference=self.hostremoval_reference,
+            priority=self.account,
+        )
+
     def config_case(
         self,
         case_id: str,
         instrument_platform: SequencingPlatform.ILLUMINA,
-        fasta: Optional[str],
+        dry_run: bool,
+        fasta: str = "",
     ) -> None:
         """Create sample sheet file for Taxprofiler analysis."""
         self.create_case_directory(case_id=case_id)
         LOG.info("Generating sample sheet")
         self.write_sample_sheet(
+            case_id=case_id, instrument_platform=instrument_platform, fasta=fasta, dry_run=dry_run
+        )
+        LOG.info("Generating parameters file")
+        self.write_params_file(
             case_id=case_id,
-            instrument_platform=instrument_platform,
-            fasta=fasta,
+            pipeline_parameters=self.get_pipeline_parameters(case_id=case_id).dict(),
+            dry_run=dry_run,
         )
-        LOG.info("Sample sheet written")
+        if dry_run:
+            LOG.info("Dry run: Config files will not be written")
+            return
```

Comment:
Done both in taxprofiler and rna-fusion.
Why shall I remove the dry-run parameters from the `write_params_file` and `write_sample_sheet`?

Reply:
These methods will exclusively be invoked from the configuration case. I find it redundant to define the dry-run logic within the methods themselves. Instead, we can move that logic to one place, which is the config case, preventing their execution from there and leading to a cleaner and more streamlined structure

Reply:
Ok, I can remove it from those two methods.
@fevac is it ok that I do the same for rna-fusion?

Reply:
Updated the PR for both taxprofiler and rnafusion. Tests are passing.

Reply:
sorry I missed this discussion. I did find it easier and cleaner with the dry-run inside as otherwise sometimes we need multiple `if dry-run` statements, specially for printing logs. But I guess it's a matter of taste?